crates/oxc_linter/src/plugin.rs

```diff
@@ -0,0 +1,233 @@
+use std::{collections::BTreeMap, env, fs, rc::Rc, sync::Arc};
+
+use oxc_diagnostics::miette::{miette, LabeledSpan};
+use oxc_query::Adapter;
+use oxc_semantic::Semantic;
+use serde::Deserialize;
+use trustfall::{execute_query, Schema, TransparentValue, TryIntoStruct};
+
+use crate::context::LintContext;
+
+enum SpanInfo {
+    SingleSpanInfo(SingleSpanInfo),
+    MultipleSpanInfo(MultipleSpanInfo),
+}
+
+#[derive(Debug, Deserialize)]
+struct SingleSpanInfo {
+    span_start: u64,
+    span_end: u64,
+}
+
+#[derive(Debug, Deserialize)]
+struct MultipleSpanInfo {
+    span_start: Box<[u64]>,
+    span_end: Box<[u64]>,
+}
+
+#[derive(Deserialize, Clone)]
+pub struct InputQuery {
+    pub name: String,
```

Comment:
the name is only ever used for printing out during testing, I wonder if we even need it... just makes the tests easier to find when they fail. maybe we could just use file path as a defacto name instead? I don't think it's that bad to keep it though

Reply:
Let's keep it for now